CreeDictionary/CreeDictionary/templates/CreeDictionary/word-entries.html

```diff
@@ -9,31 +9,77 @@ <h2 class="definition-title definition-title--search-result">
-              <span lang="cr">{{ word.text }}</span>
+            <span lang="cr">{{ normatized_cree }}</span>
+          {% endif %}
+
+              <span class="definition-title__pos">({{ lemma_wordform|presentational_pos }})</span>
+        </dfn>
+
+
+      </h2>
+    </header>
+
+      {# Show the matched lemma (when this is NOT a lemma. #}
+      {% if analysis != lemma_wordform.analysis %}
+      <p class="definition-title__reference-to-lemma" data-cy="reference-to-lemma">
+        Form of
+        {# TODO: this link should go to the lemma search result, further down the page #}
+        <a class="definition-title__link" href="javascript:void(0)" data-lemma-id="{{ lemma_wordform.id }}">{{ lemma_wordform.text }}</a>
+      </p>
+      {% endif %}
+
+    <ol class="meanings meanings--search-result">
+    {% for def in lemma_wordform.definitions.all %}
+      <li class="meanings__meaning" data-cy="lemma-meaning" data-cy="link-to-lemma">{{ def.text }}
+        {% for source in def.source_ids %}
+          <cite class="cite-dict">{{ source }}</cite>
+        {% endfor %}
+      </li>
+    {% endfor %}
+    </ol>
+  </article>
+</li>
+{% endfor %}
+
+
+{% for matched_english, normatized_cree, lemma_wordform in english_results %}
+<li class="search-results__result" data-cy="search-results">
+  <article class="definition box box--rounded" data-cy="search-result">
+    <header class="definition__header">
+      <h2 class="definition-title definition-title--search-result">
+
+        <dfn class="definition-title__word" data-cy="definition-title">
+          {# upon loaded, this <a> will be attached to onclick handler that loads #}
+          {# detailed lemma page. <a> is used for accessible styling #}
+          {% if lemma_wordform.is_lemma %}
+            <a class="definition-title__link" href="javascript:void(0)" data-lemma-id="{{ lemma_wordform.id }}">
+              {# data- attributes are supported in HTML5 not in XHTML #}
+              <span lang="cr">{{ lemma_wordform.text }}</span></a>
+          {% else %}
+              <span lang="cr">{{ normatized_cree }}</span>
           {% endif %}
         </dfn>
 
-        <span class="definition-title__pos">({{ word|presentational_pos }})</span>
+        <span class="definition-title__pos">({{ lemma_wordform|presentational_pos }})</span>
       </h2>
     </header>
 
       {# Show the matched lemma (when this is NOT a lemma. #}
-      {% if not word.is_lemma %}
-      <p class="definition-title__link-to-lemma">
+      {% if analysis != lemma_wordform.analysis %}
+      <p class="definition-title__reference-to-lemma" data-cy="reference-to-lemma">
         Form of
         {# TODO: this link should go to the lemma search result, further down the page #}
-        <a href="#">{{ word.lemma.text }}</a>
+        <a class="definition-title__link" href="javascript:void(0)" data-lemma-id="{{ lemma_wordform.id }}">{{ lemma_wordform.lemma.text }}</a>
```

Comment:
Could the `href` here link to the `id` of the lemma search result, instead of doing `javascript:void(0)`? That way, the browser will automatically scroll to the search result of the lemma (which has the actual definition).

Reply:
TIL lol, will change

Reply:
So the lemma will be like `<a id="defined:wâpamêw">wâpamêw</a>` and the matched wordform will be a link: `ê-wâpamât form of <a href="#defined:wâpamêw">wâpamêw</a>`. That way, redirection is automatically handled by the browser! No JavaScript required!

Reply:
OOooOOoh I think I misunderstood what you expected search to do here. So you expect 

> if user searches 'nipa' " we should show
> 1. nipâ - form of **nipâw**
> 2. **nipâw**
>
> and if user clicks on the first **nipâw**, it jumps to the second, but on the same page

Is that your expectation when you designed everything?

In the current search code, the second lemma is not displayed, and the first **nipâw** link actually jumps to the lemma detail page of nipâw.

Reply:
can we merge this for now, to achieve what's said requires modification to the search code. I think it will be easy to add on, based on the current detailedly typed search code.